dev-client/src/store/persistence.test.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {configureStore, createSlice} from '@reduxjs/toolkit';
+
+const {
+  reducer,
+  actions: {increment},
+} = createSlice({
+  name: 'test',
+  initialState: {counter: 0},
+  reducers: {
+    increment: ({counter}) => ({counter: counter + 1}),
+  },
+});
+
+test('persistence middleware saves state to disk', () => {
+  jest.isolateModules(() => {
+    const {kvStorage} = require('terraso-mobile-client/persistence/kvStorage');
+    kvStorage.setBool('FF_offline', true);
+
+    const {
+      persistenceMiddleware,
+      loadPersistedReduxState,
+    } = require('terraso-mobile-client/store/persistence');
```

Comment:
Question: Could you explain the use of `require` and how it differs from import here?

Reply:
answers:
- you can only use `import` at the top level of a file, not within a callback, otherwise i would use `import`
- yeah exactly! MMKV has functionality to reset the store after each test, but
- the `terraso-mobile-client/store/persistence` module imports the `terraso-mobile-client/config/featureFlags` module, so we need to re-initialize it on each test to be able to control the state of the feature flags

i can see this getting out of hand if we're testing lots of functionality that's gated behind feature flags, but this works for now and we can always do something smarter if it turns into a big DX issue

Reply:
Would it help to add to featureFlags.ts something like `enableFlag_forTestOnly`, that would add a given flag to `ENABLED_FLAGS` and allow us to bypass the feature flag-related kvStorage stuff?